src/components/Editor/Attachments/AttachmentsList.vue

```diff
@@ -0,0 +1,222 @@
+<template>
+	<div id="attachments">
+		<input ref="localAttachments"
+			type="file"
+			multiple
+			style="display:none"
+			@change="onLocalAttachmentSelected">
+		<div v-if="attachments.length > 0">
+			<ul>
+				<ListItem v-for="attachment in attachments"
+					:key="attachment.value"
+					:force-display-actions="true"
+					:title="attachment.fileName">
+					<template #icon>
+						<Paperclip :size="24" />
```

Comment:
As an improvement, we *should* be able to show the file preview here instead.

Reply:
@tcitworld  That would be really great! However, I don't really know =( yet how to display file previews (it would also be relevant for downloading attachments in the mail application).  

I know that links like " are generated http://nextcloud.dev.loc/index.php/core/preview?fileId=607478&x=250&y=250&forceIcon=0&a=0"  

I read the description here, but there was a question of generating office documents. 

https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/previews_configuration.html 

**I would be grateful for a hint in this direction.**

Reply:
I did not find a cleaner way than https://github.com/nextcloud/viewer/blob/43e7003b3095bba553d71f59e73aa8ec1dbdff29/src/mixins/PreviewUrl.js#L57-L82

We should have this exacted in a new package.

Reply:
**Thank you very much. I will apply this method.**

That is, do I understand correctly that if I don't have a `collabora + nextcloud office` bundle, then I get previews only for pictures, mp3 and text files? There is also no way to get a preview at the time of download. 

That's all I'm getting at: I probably need to replace the missing images with "something" (file icons, for example).
BTW
There are still such addresses.
`/index.php/apps/theming/img/core/filetypes/x-office-document.svg`
Probably I should also look for how icons are generated according to the mime type.

Reply:
https://github.com/nextcloud/server/blob/fd5f7486da8192b6ebf69ee3e6640cb7662028ca/apps/files/src/views/Sidebar.vue#L310-L337
As I understand it, so far there is no single way (or component) to get a preview or icon

Reply:
In general, I had to use `MANAGED-ID` and `split()` magic =)
Perhaps there are better solutions. But I can only see this one so far.
![image](https://user-images.githubusercontent.com/3595562/172137159-4589d30d-8624-432e-a84b-9eae87d34268.png)

_In general, the topic of the preview is quite ticklish_